core/domain/stats_domain.py

```diff
@@ -1351,3 +1365,360 @@ def validate(self):
+        self.answer = answer
+        self.answer_details = answer_details
+        self.created_on = created_on
+
+    def to_dict(self):
+        """Returns the dict of learner answer info.
+
+        Returns:
+            dict. The learner_answer_info dict.
+        """
+        learner_answer_info_dict = {
+            'id': self.id,
+            'answer': self.answer,
+            'answer_details': self.answer_details,
+            'created_on': str(self.created_on)
+        }
+        return learner_answer_info_dict
+
+    @classmethod
+    def from_dict(cls, learner_answer_info_dict):
+        """Returns a dict representing LearnerAnswerInfo domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of LearnerAnswerInfo instance.
+        """
+
+        return cls(
+            learner_answer_info_dict['id'],
+            learner_answer_info_dict['answer'],
+            learner_answer_info_dict['answer_details'],
+            datetime.datetime.strptime(
+                learner_answer_info_dict['created_on'], '%Y-%m-%d %H:%M:%S.%f')
+        )
+
+    @classmethod
+    def get_learner_answer_info_id(cls, state_reference, entity_type):
+        """Generates the learner answer info domain object id.
+
+        Args:
+            state_reference: str. This is used to refer to a state
+                in an exploration or question. Like for an exploration the
+                value will be equal to 'exp_id.state_name' & for question
+                this will be equal to 'question_id' only.
+            entity_type: str. The type of entity i.e 'exploration' or
+                'question '.
+
+        Return:
+            learner_answer_info_id: str. The id generated by the function.
+        """
+        learner_answer_info_id = (state_reference + '.' + entity_type + '.' +
+                                  utils.base64_from_int(
+                                      utils.get_current_time_in_millisecs()) +
+                                  utils.base64_from_int(
+                                      utils.get_random_int(127 * 127)))
+        return learner_answer_info_id
+
+    def validate(self):
+        """Validates the LearnerAnswerInfo domain object."""
+
+
+        if not isinstance(self.id, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.id)
+        if self.answer is None:
+            raise utils.ValidationError(
+                'The answer submitted by the learner cannot be empty')
+        if self.answer_details is None:
+            raise utils.ValidationError(
+                'LearnerAnswerInfo must have a provided answer details')
+        if sys.getsizeof(self.answer_details) > MAX_ANSWER_DETAILS_BYTE_SIZE:
+            raise utils.ValidationError('The answer details size is to large '
+                                        'to be stored')
+        if not isinstance(self.answer_details, basestring):
```

Comment:
This check will fail if it's None, right? I'd suggest removing the None check above.

Reply:
Done